From 5c3683f28a878759544536f80735ebe0f1f57dd8 Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Fri, 10 Sep 2021 09:13:41 +0800
Subject: [PATCH] Grant runtime permissions by default

Based on the customer requirement, we need to disable the
permission grant pop-up window.

Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 .../android/server/pm/permission/PermissionManagerService.java  | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/services/core/java/com/android/server/pm/permission/PermissionManagerService.java b/services/core/java/com/android/server/pm/permission/PermissionManagerService.java
index 8d2363b6e831..0b50e353956f 100644
--- a/services/core/java/com/android/server/pm/permission/PermissionManagerService.java
+++ b/services/core/java/com/android/server/pm/permission/PermissionManagerService.java
@@ -2845,6 +2845,8 @@ public class PermissionManagerService extends IPermissionManager.Stub {
                             + friendlyName);
                 }
 
+                grant = GRANT_INSTALL;
+
                 if (grant != GRANT_DENIED) {
                     if (!ps.isSystem() && ps.areInstallPermissionsFixed() && !bp.isRuntime()) {
                         // If this is an existing, non-system package, then
-- 
2.25.1

